telemetry/record.go

```diff
@@ -127,9 +128,46 @@ func (record *Record) injectPayloadWithVin() error {
 			return err
 		}
 		message.Vin = record.Vin
+		transformLocations(message)
 		record.PayloadBytes, err = proto.Marshal(message)
 		return err
 	default:
 		return nil
 	}
 }
+
+// transformLocations does a best-effort attempt to convert the Location field to a proper protos.Location
+// type if what we receive is a string that can be parsed. This should make the transition from strings to
+// Locations easier to handle downstream.
+func transformLocations(message *protos.Payload) {
+	for _, datum := range message.Data {
+		if datum.GetKey() == protos.Field_Location && datum.GetValue().GetStringValue() != "" {
+			if loc, err := ParseLocation(datum.GetValue().GetStringValue()); err == nil {
+				datum.Value = &protos.Value{Value: &protos.Value_LocationValue{LocationValue: loc}}
```

Comment:
Since there are at max one datum with `protos.Field_Location `, can we return here transformation is done once?

Reply:
Hmm, technically we could have more than one, no? Or do we specify somewhere that in no case can there be any `Datum` with the same key in a given `Payload`?

Reply:
each array can only have unique `Key`

Reply:
Made it exit the loop after the Location field is found. Made the loop a teeny bit more complex as the first `if` had to be split into two.